force-app/main/default/classes/SOQL.cls

```diff
@@ -85,9 +118,26 @@ public with sharing class SOQL {
      * @return A map of Task Ids to Task records.
      * Update the corresponding test class and testGetTasksDueNextWeek method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
+    public static Map<Id, Task> getTaskDueNextWeek() {
+
+        Map<Id,Task> taskMap = new Map<Id, Task>();
+        
+        List<Task> taskDueDate = [SELECT Id, ActivityDate FROM Task WHERE ActivityDate = NEXT_N_DAYS:7];
```

Comment:
There is a special constructor for Map<Id, SObject> that takes a List or query result and separates it into a map of the Id to the record itself.